diff --git a/render/vulkan/vulkan.c b/render/vulkan/vulkan.c
index 7e9e9396..42e98b91 100644
--- a/render/vulkan/vulkan.c
+++ b/render/vulkan/vulkan.c
@@ -164,7 +164,7 @@ struct wlr_vk_instance *vulkan_instance_create(size_t ext_count,
 	application_info.apiVersion = VK_API_VERSION_1_1;
 
 	const char *layers[] = {
-		"VK_LAYER_KHRONOS_validation",
+		// "VK_LAYER_KHRONOS_validation",
 		// "VK_LAYER_RENDERDOC_Capture",
 		// "VK_LAYER_live_introspection",
 	};
